sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1774,6 +1775,34 @@ class Analyzer(
   }
 
   /**
+   * Substitute Hints.
+   */
+  object SubstituteHint extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan transform {
```

Comment:
Still quite curious what will happen here:
```sql
SELECT /*+ MAPJOIN(tbl_b) */
       *
FROM   tbl_a A
       LEFT JOIN tbl_b B
        ON B.id = A.id
       LEFT JOIN (SELECT XA.id
                  FROM   tbl_c XA
                         LEFT ANTI JOIN tbl_b XB
                          ON XB.id = XA.id) C
        ON C.id = A.id
```
Could you add some documentation for this here?